### DISCUSSION TITLE
Select GPU from cli

### DISCUSSION THREAD

**Discussion by DuaneNielsen** [Lightning-AI/pytorch-lightning#7461] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
The CLI has a flag --gpus

In a system with more than 1 GPU, is there a way to select the GPU you want to run on from CLI?

I tried --gpus [1] to select cuda:1 but it doesn't work.

Also the auto gpu selection didn't work for me.  It tried to put the job on cuda:0, but cuda:0 didn't have enough memory to run it.

in the end I resorted to CUDA_VISIBLE_DEVICES, but that seems silly.

Thanks!

**Comment by carmocca** [comment, marked as answer]
I'm assuming you are referring to the `LightningCLI`

In that case, just do `python yourscript.py --trainer.gpus=[1]`

**Comment by tshu-w** [comment]
for those who use zsh, maybe we need `\[1\]`

**Reply by carmocca** [reply]
Alternatively, `--trainer.gpus='[1]'`

**Reply by tshu-w** [reply]
Thx, it looks nicer.

**Reply by DuaneNielsen** [reply]
Since I was using 

```
import pytorch_lightning as pl
from argparse import ArgumentParser

parser = ArgumentParser()
parser = pl.Trainer.add_argparse_args(parser)
args = parser.parse_args()

trainer = pl.Trainer.from_argparse_args(args)
```

I ended up doing 

```
my_program.py -gpus 1,
```

The comma caused pl to interpret the argument as a list of gpu device ids instead of the number of gpus.
